impl/src/main/java/impl/AnswerResourceImpl.java

```diff
@@ -71,14 +75,14 @@ public AnswerResourceImpl(AnswerDao answerDao, QuestionDao questionDao, DaoTrans
 
     @Override
     public Observable<Void> markAsAcceptedAnswer(Auth auth, long answerId) {
-
-        return answerDao.getQuestionIdByAnswer(answerId)
-            .flatMap(questionId -> {
-                Observable<Integer> markAnswerAsAccepted = answerDao.markAsAccepted(answerId);
-                Observable<Integer> markQuestionAsAccepted = questionDao.markAsAnswered(auth.getUserId(), questionId);
-
-                List<Observable<Integer>> acceptances = asList(markAnswerAsAccepted, markQuestionAsAccepted);
-                return this.daoTransactions.executeTransaction(acceptances);
+        return answerDao.getAnswerById(answerId)
+            .flatMap(answer -> {
+                if (answer.getQuestion().getUserId() != auth.getUserId()) {
+                    return exception(() -> new WebException(BAD_REQUEST, ERROR_NOT_OWNER_OF_QUESTION));
+                }
+                Observable<Integer> markAnswerAsAccepted   = answerDao.markAsAccepted(answerId);
+                Observable<Integer> markQuestionAsAccepted = questionDao.markAsAnswered(auth.getUserId(), answer.getQuestionId());
```

Comment:
here we have a terminology issue :) accepted or answered? I think that answered is more correct from a questions perspective. But i might very well be wrong :)

Reply:
True.